src/server.py

```diff
@@ -0,0 +1,106 @@
+# coding=utf-8
+import io
+import mimetypes
+import os
+import socket
+
+
+def server():
+    """Run Server."""
+    server_socket = socket.socket(socket.AF_INET,
+                                  socket.SOCK_STREAM,
+                                  socket.IPPROTO_TCP)
+    server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    address = ('127.0.0.1', 5000)
+    server_socket.bind(address)
+    server_socket.listen(1)
+    conn, addr = server_socket.accept()
+
+    buffer_length = 8
+    message_complete = False
+    return_message = ''
+    while not message_complete:
+        part = conn.recv(buffer_length)
+        if len(part) < buffer_length:
+            message_complete = True
+            server_socket.close()
+        return_message += part.decode('utf-8')
+    print(return_message)
+    try:
+        uri = parse_request(return_message)
+        content = resolve_uri(uri)
+        reply_ok = response_ok()
+        content_type = b'Content Type: ' + content[1] + b'\r\n'
+        content_length = str(len(content[0])).encode('utf-8')
+        content_length = b'Content-Length: ' + content_length + b'\r\n\r\n'
+        reply = reply_ok + content_type + content_length + content[0]
```

Comment:
What if you were to build a list of strings instead and then join them all together with `'\r\n'`:
```python
lines = [
    b'HTTP/1.1 200 OK',
    b'Content-Type: %s' % content[1],
    b'Another-Header: foo',
    b'',
    content[0]
]
b'\r\n'.join(lines)
```